editor/editor_resource_preview.cpp

```diff
@@ -441,27 +456,38 @@ void EditorResourcePreview::check_for_invalidation(const String &p_path) {
 }
 
 void EditorResourcePreview::start() {
-	if (DisplayServer::get_singleton()->get_name() != "headless") {
+	if (DisplayServer::get_singleton()->get_name() == "headless") {
+		return;
+	}
+
+	if (RSG::texture_storage->can_create_resources_async()) {
 		ERR_FAIL_COND_MSG(thread.is_started(), "Thread already started.");
 		thread.start(_thread_func, this);
+	} else {
+		SceneTree *st = Object::cast_to<SceneTree>(OS::get_singleton()->get_main_loop());
+		if (st) {
```

Comment:
Is there any realistic scenario where the editor main loop wouldn't be SceneTree?
If not, we might want to put a `ERR_FAIL_NULL(st);` there for good measure instead of `if (st)`.

Reply:
Good point. Changed.